### ISSUE TITLE
TypeError: 'NoneType' object is not iterable - 500 errors in Webschemas log

### ISSUE THREAD

**Issue by RichardWallis** [schemaorg/schemaorg#1400]
Fairly common error in logs of webschemas:

TypeError: 'NoneType' object is not iterable
File "/base/data/home/apps/s~webschemas-g/1.396259617206838128/sdoapp.py", line 707, in ClassProperties
subprops = sorted(prop.subproperties(layers=layers),key=lambda u: u.id)

**Comment by danbri** [comment]
Can you summarize the fix?

**Comment by RichardWallis** [comment]
Analysing the code at the point of failure, there is potential that a sort operation will be given a list of potentially null items.

Made a couple of tweaks to 1. delegate the sorting to a later stage and only if a non-null list is returned.  2. Removed the unnecessary creation of some similar lists that are then not used within the function in question.

**Comment by RichardWallis** [comment]
Closed - now covered under issue #2060
